### PR TITLE
Update iotago and Cluster tests

### PR DESCRIPTION
**Pull request by lmoe**
# Description of change

* Updates iota.go/v3 to `946b36ac8ab6`
* Refactoring of the project
* Updates to EVM binaries
* Updates the cluster tests and private tangle to use recent inx/hornet nodes

## How the change has been tested

make test-short
make test

### CODE REVIEW

File: packages/testutil/privtangle/privtangle.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func Start(ctx context.Context, baseDir string, basePort, nodeCount int, logfunc
 		NodeKeyPairs:  make([]*cryptolib.KeyPair, nodeCount),
 		NodeCommands:  make([]*exec.Cmd, nodeCount),
 		ctx:           ctx,
+		// logfunc:       logfunc,
```
Comment (dessaya) [inline packages/testutil/privtangle/privtangle.go]: Is this intended?

Comment (jorgemmsilva) [inline packages/testutil/privtangle/privtangle.go]:
```suggestion
```

File: packages/testutil/privtangle/privtangle.go [hunk 2]
```diff
@@ -161,7 +163,6 @@ func (pt *PrivTangle) startNode(i int) {
 		"-c", pt.ConfigFile,
 		fmt.Sprintf("--protocol.parameters.networkName=%s", pt.NetworkName),
 		fmt.Sprintf("--restAPI.bindAddress=0.0.0.0:%d", pt.NodePortRestAPI(i)),
-		fmt.Sprintf("--dashboard.bindAddress=localhost:%d", pt.NodePortDashboard(i)),
```
Comment (jorgemmsilva) [inline packages/testutil/privtangle/privtangle.go:164]: any reason to remove this?

**Review by dessaya — APPROVED** [review]

**Review by jorgemmsilva — APPROVED** [review]

File: packages/testutil/privtangle/privtangle.go (hunk 2 above)
Comment (lmoe) [reply packages/testutil/privtangle/privtangle.go:164]: Yeah the node quits when you add this argument. I think it was moved to the inx-dashboard node.

**Review by Juliusan — APPROVED** [review]
I see no breaking changes in consensus. However, as `MilestoneIndex` got deleted from `iscp.TimeData` structure, there is no need to keep the whole structure. I would recommend using `time.Time` instead of `iscp.TimeData` everywhere. Unless, it is expected that milestones are going to need a structure once again.

**Comment by lmoe** [comment]
Alright. Let me put this into another PR though, it's pretty huge. :D

**Pull request merged**